### PR TITLE
fix: signup buttons

### PR DESCRIPTION
**Pull request by daibhin**
## Problem

We need a stronger "CTA" style button for places like signup & onboarding

## Changes

- Introduce new "onboarding" style button that matches the primary button from the marketing site
- Make sure "onboarding" replicates styles of primary in the legacy UI

### `onboarding` looks like primary button in legacy UI
Screenshot 2023-11-23 at 11 28 59

### `onboarding` button in light and dark mode next to classic "primary" button
|Light|Dark|
|----|----|
| Screenshot 2023-11-23 at 11 39 02 | Screenshot 2023-11-23 at 11 39 10 |

**Note:** The CTA button has the same vibrant orange in light and dark mode. We could use the more "subtle" orange in dark mode but that's not what is done on the website. I think it's fine for now because the onboarding will not support dark mode initially.

### Onboarding changes
|Before|After|
|----|----|
| Screenshot 2023-11-23 at 11 32 26 | Screenshot 2023-11-23 at 11 36 05 |

|Before|After|
|----|----|
| Screenshot 2023-11-23 at 11 32 30 | Screenshot 2023-11-23 at 11 22 48 |

|Before|After|
|----|----|
| Screenshot 2023-11-23 at 11 32 44 | Screenshot 2023-11-23 at 11 30 22 |

Most important change:
|Before|After|
|----|----|
| Screenshot 2023-11-23 at 09 50 32 | Screenshot 2023-11-23 at 09 51 17 |

## How did you test this code?

- Screenshots & snapshots

### CODE REVIEW

**Comment by daibhin** [comment]
@raquelmsmith @benjackwhite I only changed the buttons screenshotted in the PR description but let me know if I've missed any other places

File: frontend/src/lib/lemon-ui/LemonButton/LemonButton3000.scss [hunk 1]
```diff
@@ -152,7 +153,23 @@
             }
         }
 
+        &.LemonButton--cta {
+            --lemon-button-border-width: 1px; // TODO: increase this to 1.5px once snapshotting is in place to make sure other buttons are not affected
+
+            background: #cd8407;
```
Comment (benjackwhite) [inline frontend/src/lib/lemon-ui/LemonButton/LemonButton3000.scss:159]: Hardcodded var here and below? Is that necessary?

**Review by benjackwhite — COMMENTED** [review]
Tagging @Twixes has he knows the button code much better

**Review by Twixes — CHANGES_REQUESTED** [review]
The visuals are great, but I'm pretty confused by the API. The thinking with `type` vs `status` was that the type is basically the shape, while the status is the color – as evidenced by the LemonButton stories. This "cta" type is basically primary, except slightly differently colored, right? In that case it should just be a status.

This is important, because we need everyone at PostHog to have a trivially easy time using the right UI for what they're building, and so every type/status combination should be reasonably distinct and obvious. I don't think this is the case with "primary" vs. "cta":

Screenshot 2023-11-23 at 14 27 29

**Comment by Twixes** [comment]
Looking at the website, if we're to match its dark mode style, we already have the "cta" status in dark mode in the form of "primary". We don't however have buttons like the one on the right in the app.
Screenshot 2023-11-23 at 14 30 04

**Comment by daibhin** [comment]
@Twixes I totally agree with how you're thinking about what the API should be but don't know if it's possible without a big refactor worth of changes needed to make everything clearer. 

The way things are implemented right now is that we don't rely on `status` at all in 3000. Our `primary` and `secondary` button are identical in shape / function and yet are denoted by `type` not `status`.

I'd suggest waiting until the old UI styles have all been removed and everything will be much easier to cleanup because we don't have to worry about what the buttons map to in all the legacy cases they're used in. My thinking on evolving the button API to clean everything up would be:
- [Today] Add `"onboarding"` as a button `type` so that it's a small isolated change
- [Release 3000] to everyone and remove all the old legacy styles
- Deprecate what is now the `"secondary"` button type and make it a `"status"`
- Deprecate the `"onboarding"` type and also make it a `"status"` on the `"primary"` button type
- [Open question] Make stealth a variant or the new `"secondary"`. Depends on whether we want to allow the primary button to be stealth too.

I renamed "cta" to "onboarding" for now in a hope to make it clearer during the transition period that you probably don't need this button type.

> Looking at the website, if we're to match its dark mode style, we already have the "cta" status in dark mode in the form of "primary". We don't however have buttons like the one on the right in the app.

[I asked Cory about this too](https://posthog.slack.com/archives/C064U4XCPHQ/p1700745437073309?thread_ts=1700734370.691169&cid=C064U4XCPHQ). I agree it feels weird from a system point of view to have a style that looks identical to the "primary" style (in dark mode), irrespective of how we denote it (calling it "status" or "type"). His feeling is that the button on the right is just not prominent enough so we don't necessarily want it in the app.

**Comment by daibhin** [comment]
Sorry for such a long comment 😅 Just wanted to leave as much context as possible before I head off tomorrow

**Comment by Twixes** [comment]
Thanks for the context. Makes sense! Let's get this properly in order later. (Although one thing we should 100% do for 3000 before rollout is support for the danger status.)

**Review by Twixes — APPROVED** [review]

**Pull request merged**